### PR TITLE
Fix devel compile errors when using centos/RH 7.2 os

### PR DESCRIPTION
**Pull request by leibin2014**
## What
Fix devel compile errors when using centos/RH 7.2 os

## Why ?
When using centos/RH 7.2 os some errors can caused by undefined macros.

## How ?
Replace with more general code style.

### CODE REVIEW

File: src/tools/perf/lib/uct_tests.cc [hunk 1]
```diff
@@ -506,8 +506,7 @@ class uct_perf_test_runner {
                     /* Wait until getting ACK from responder */
                     sn = get_recv_sn(recv_sn, m_perf.uct.recv_mem.mem_type);
                     ucs_assertv(UCS_CIRCULAR_COMPARE8(send_sn - 1, >=, sn),
-                                "recv_sn=%d iters=%" PRIu64, sn,
-                                m_perf.current.iters);
+                                "recv_sn=%d iters=%lu", sn, m_perf.current.iters);
```
Comment (yosefe) [inline src/tools/perf/lib/uct_tests.cc:509]: "iters" is uint64_t, so better use PRIu64
can we include `<inttypes.h>` as a fix?

File: src/ucs/datastruct/bitmap.h [hunk 2]
```diff
@@ -536,7 +536,7 @@ ucs_bitmap_ffs(const ucs_bitmap_word_t *bitmap_words, size_t num_words,
             return _UCS_BITMAP_BIT_INDEX(first_bit_in_word, word_index);
         }
 
-        mask = UINT64_MAX;
+        mask = ((uint64_t) -1);
```
Comment (yosefe) [inline src/ucs/datastruct/bitmap.h:539]: UCS_MASK_SAFE(UCS_BITMAP_BITS_IN_WORD)

File: src/tools/perf/lib/uct_tests.cc (hunk 1 above)
Comment (leibin2014) [reply src/tools/perf/lib/uct_tests.cc:509]: @yosefe Both the customer and I tried this approach before but it didn't help.

File: src/ucs/datastruct/bitmap.h (hunk 2 above)
Comment (leibin2014) [reply src/ucs/datastruct/bitmap.h:539]: @yosefe I got below comple error when I change to this:
```In file included from /root/bilei/ucx/contrib/../src/ucs/memory/memory_type.h
                 from /root/bilei/ucx/contrib/../test/gtest/common/mem_buffer
                 from /root/bilei/ucx/contrib/../test/gtest/common/mem_buffer
/root/bilei/ucx/contrib/../src/ucs/datastruct/bitmap.h: In function 'int ucs_t*, size_t, size_t)':
/root/bilei/ucx/contrib/../src/ucs/sys/compiler_def.h:74:44: error: left shif]
 #define UCS_BIT(i)               (1ul << (i))
                                            ^
/root/bilei/ucx/contrib/../src/ucs/sys/compiler_def.h:77:35: note: in expansi
 #define UCS_MASK(i)              (UCS_BIT(i) - 1)
                                   ^
/root/bilei/ucx/contrib/../src/ucs/sys/math.h:93:40: note: in expansion of ma
     (((_i) >= 64) ? ((uint64_t)(-1)) : UCS_MASK(_i))
                                        ^
/root/bilei/ucx/contrib/../src/ucs/datastruct/bitmap.h:540:16: note: in expan
         mask = UCS_MASK_SAFE(UCS_BITMAP_BITS_IN_WORD);```

Simplifying the error is caused by the code "1ul << 64".

Comment (leibin2014) [reply src/ucs/datastruct/bitmap.h:539]: Bypassed the error by changing to:
···
        size_t bits;
        bits = UCS_BITMAP_BITS_IN_WORD;
        mask = UCS_MASK_SAFE(bits);
···

**Comment by yosefe** [comment]
replaced by #6607

**Pull request closed without merging**